starlite/app.py

```diff
@@ -243,68 +244,107 @@ def __init__(
+            before_shutdown=before_shutdown or [],
+            cache_config=cache_config,
+            compression_config=compression_config,
+            cors_config=cors_config,
+            csrf_config=csrf_config,
+            debug=debug,
+            dependencies=dependencies or {},
+            exception_handlers=exception_handlers or {},
+            guards=guards or [],
+            middleware=middleware or [],
+            on_shutdown=on_shutdown or [],
+            on_startup=on_startup or [],
+            openapi_config=openapi_config,
+            parameters=parameters or {},
+            plugins=plugins or [],
             response_class=response_class,
-            response_cookies=response_cookies,
-            response_headers=response_headers,
+            response_cookies=response_cookies or [],
+            response_headers=response_headers or {},
             route_handlers=route_handlers,
-            security=security,
-            tags=tags,
+            security=security or [],
+            static_files_config=static_files_config or [],
+            tags=tags or [],
+            template_config=template_config,
+        )
+        for handler in on_app_init or []:
+            config = handler(config)
+
+        self.after_exception = as_async_callable_list(config.after_exception)
+        self.after_shutdown = as_async_callable_list(config.after_shutdown)
+        self.after_startup = as_async_callable_list(config.after_startup)
+        self.allowed_hosts = config.allowed_hosts
+        self.before_send = as_async_callable_list(config.before_send)
+        self.before_shutdown = as_async_callable_list(config.before_shutdown)
+        self.before_startup = as_async_callable_list(config.before_startup)
+        self.cache = config.cache_config.to_cache()
+        self.compression_config = config.compression_config
+        self.cors_config = config.cors_config
+        self.csrf_config = config.csrf_config
+        self.debug = config.debug
+        self.on_shutdown = config.on_shutdown
+        self.on_startup = config.on_startup
+        self.openapi_config = config.openapi_config
+        self.plugins = config.plugins
+        self.static_files_config = config.static_files_config
+        self.template_engine = create_template_engine(config.template_config)
+        super().__init__(
+            after_request=config.after_request,
+            after_response=config.after_response,
+            before_request=config.before_request,
+            dependencies=config.dependencies,
+            exception_handlers=config.exception_handlers,
+            guards=config.guards,
+            middleware=config.middleware,
+            parameters=config.parameters,
+            path="",
+            response_class=config.response_class,
+            response_cookies=config.response_cookies,
+            response_headers=config.response_headers,
+            route_handlers=config.route_handlers,
```

Comment:
```suggestion
            route_handlers=[],
```
See https://github.com/starlite-api/starlite/pull/499#discussion_r978227790

Reply:
tests pass as we go on to register them after the `Router.__init__()` call.